tests/unit/db/test_ingest.py

```diff
@@ -26,55 +37,198 @@ def test_cve_data_ingested(self, cve_examples: Path) -> None:
         for cve in nvddb.cve_all():
             assert isinstance(cve, CVE)
 
-    @pytest.mark.parametrize(
-        ("offset", "last_mod_start_date"),
-        [
-            (None, None),
-            (0, None),
-            (42, None),
-            (None, datetime.now(timezone.utc)),
-            (42, datetime.now(timezone.utc)),
-        ],
-    )
-    def test_cve_api_args(self, offset: int | None, last_mod_start_date: datetime | None) -> None:
+    @pytest.mark.parametrize("update", [False, False, False, True, True])
+    def test_cve_api_args(self, update: bool) -> None:
         with patch("bomsquad.vulndb.db.ingest.NVD.vulnerabilities") as vulns:
-            if offset:
-                Ingest.cve(offset, last_mod_start_date=last_mod_start_date)
-            else:
-                Ingest.cve(last_mod_start_date=last_mod_start_date)
+            Ingest.cve(update=update)
             assert vulns.call_count == 1
             args, kwargs = vulns.call_args
-            if offset:
-                assert args[0] == offset
+            assert kwargs["offset"] == 0
+            cp = Checkpoints()
+            if update:
+                assert kwargs["last_mod_start_date"] == cp.last_updated("cve")
             else:
-                assert args[0] == 0
-            assert kwargs["last_mod_start_date"] == last_mod_start_date
+                assert kwargs["last_mod_start_date"] is None
 
```

Comment:
I'm thinking that we should work in a cve_update_examples and cpe_update_examples into test_data in conftest.py to simulate the update. That's kind of the ingest test driver, and then we verify we collected all of the expected records here.